src/test/spec/unified_runner/matcher.rs

```diff
@@ -154,6 +163,57 @@ fn cmap_events_match(actual: &CmapEvent, expected: &ExpectedCmapEvent) -> Result
     }
 }
 
+fn sdam_events_match(
+    actual: &SdamEvent,
+    expected: &ExpectedSdamEvent,
+    description: impl AsRef<str>,
```

Comment:
taking a description here is a step towards having more context aware failures, but I limited the changes to mostly just here. We'll want to make a more comprehensive effort for that in later PRs.